types/react-dom/client.d.ts

```diff
@@ -32,11 +32,25 @@ export interface Root {
 }
 
 /**
- * Replaces `ReactDOM.render` when the `.render` method is called and enables Concurrent Mode.
+ * Different release channels declare additional types of ReactNode this particular release channel accepts.
+ * App or library types should never augment this interface.
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_CREATE_ROOT_CONTAINERS {}
+
+export type Container =
+    | Element
+    | DocumentFragment
+    | DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_CREATE_ROOT_CONTAINERS[
+        keyof DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_CREATE_ROOT_CONTAINERS
+    ];
```

Comment:
`DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_CREATE_ROOT_CONTAINERS ` will not leak into error messages or type hints. TypeScript will resolve `T[keyof T]`:

![Screenshot 2024-03-04 at 23 27 45](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/12292047/73f19602-829a-4a60-8bd5-eedd83e548a0)

Reply:
whoa that's clever